### ISSUE TITLE
Intermittent failing test, Kafka + remote doc store

### ISSUE THREAD

**Issue by jarohen** [xtdb/xtdb#682]
```
lein test :only crux.api-test/test-adding-back-evicted-document

FAIL in (test-adding-back-evicted-document) (api_test.clj:211)
Kafka and Remote Doc Store
expected: (api/entity (api/db *api*) :foo)
actual: (not (api/entity ... :foo))
```

https://circleci.com/gh/juxt/crux/4048

**Comment by jarohen** [comment]
@jonpither - any ideas?

**Comment by t-taylor** [comment]
TL;DR Only broken because mem doc store used in testing, hence not a priority. Might be fixed with the new doc store work, so waiting on that.

Running the test on a standard kafka node with embedded kafka doesn't reproduce the issue (script [here](https://gist.github.com/t-taylor/c7506ec1ea42f052b7c2c3a854e9416f)). However the testing suite uses an in [memory document store](https://github.com/juxt/crux/blob/master/crux-test/test/crux/fixtures/doc_store.clj) and with this addition, it cause the test to fail around 6% of the time.
Chasing this further we tried to see if we could determine if the ordering of the ingest and evicting caused this issue due to a race condition by printing out the order which the instructions had been sent/received. However, creating a loop to repeat the test 100 time failed on the 9th run consistently. This was due to how clojure's [`IPersistantMap` works](https://clojuredocs.org/clojure.core/array-map#example-57392e25e4b071da7d6cfd0c). Turns out our internal `#crux/id` equality operator was inconsistent (Fix [here](https://github.com/juxt/crux/commit/f99482f8de04824ef7bc9a194c78384ff3d832ae)). With further testing, we're pretty sure that this bug is only when the memory doc store is used, which is only done in test. So for the moment this bug is going to be moved back to backlog until the current doc store work is finished as this might fix it :crossed_fingers:.

**Comment by jarohen** [comment]
Have replaced the `InMemDocumentStore` test fixture with the main standalone doc store (see the linked commit), let's see if that fixes things

**Comment by jarohen** [comment]
haven't seen a failure on this since the migration to the standalone doc store, closing for now
